File: java/src/main/java/it/giannibombelli/wsc2026/booking/domain/booking/BookingId.java
package it.giannibombelli.wsc2026.booking.domain.booking;

import it.giannibombelli.wsc2026.common.utils.Require;

import it.giannibombelli.wsc2026.common.domain.identity.AggregateId;

import java.util.UUID;

public record BookingId(UUID value) implements AggregateId {
    public BookingId {
        Require.requireArgument(value, "BookingId value");
    }
}

File: typescript/src/booking/domain/booking/bookingId.ts
// Identità dell'aggregato Booking.

import { requireArgument } from '@/common/utils/requireArgument.js';
import type { AggregateId } from '@/common/domain/identity/aggregateId.js';
import { Uuid } from '@/common/domain/primitive/uuid.js';

export class BookingId implements AggregateId {
  readonly value: Uuid;

  constructor(value: Uuid) {
    requireArgument(value, 'BookingId value');
    this.value = value;
  }
}
